Slim/Utils/Prefs/Base.pm

```diff
@@ -92,7 +92,7 @@ sub set {
 	my $namespace = $root->{'namespace'};
 	my $clientid  = $class->{'clientid'} || '';
 
-	if (!ref $new && defined $new && defined $old && $new eq $old) {
+	if ( !ref $new && ( (defined $new && defined $old && $new eq $old) || (!(defined $new) && !(defined $old)) ) ) {
```

Comment:
Wouldn't the following do the same and be easier on the eyes?

```perl
if ( (!defined $new && !defined $old) || (!ref $new && defined $new && defined $old && $new eq $old) ) {
```

Reply:
I read the Perldoc entry for `defined` and was left unsure about how safe it would be to use it on an array/hash, so thought it best to continue with the `!ref` check in all cases.

Reply:
Prefs are always scalars - or refs. An undefined reference is just undefined. It's not the same as a reference to an empty array.